### ISSUE TITLE
Complex screen objects - Question

### ISSUE THREAD

**Issue by Sebboulnois** [mousebird/WhirlyGlobe#994]
Hello everyone,

A few days ago, I created a "complex" MaplyBillboard object composed of 2 images (MaplyScreenObjects) following those steps:

1) I created the first MaplyScreenObject (OBJ1) composed of an image;
2) I created the second MaplyScreenObject (OBJ2) also compose of an image;
3) I added the OBJ2t in OBJ1 by doing OBJ1.add(OBJ2);
4) I finally created a MaplyBillBoard and associated OBJ1 (compose of 2 images) to it by doing billBoard.screenObj = OBJ1;

The result is quite good:

![pictogramowsas](https://user-images.githubusercontent.com/15698398/30031079-17477008-915e-11e7-835b-734b32767a37.jpg)

My first question is: 

Is it the most efficient way to create a MaplyBillBoard composed of 2 images?

My second question concerns the same concept in a map.
I wanted to form a complex object composed of either 2 images of 1 image and 1 label. However, it looks like neither MaplyScreenMarker nor MaplyScreenLabel have the screenObj property. 

How can I then build this complex object in 2D?

Have a nice day,

Sebastien.

**Comment by mousebird** [comment]
Yes, that's a good way to make the billboard.  I believe ScreenObjects can also take text which is even more efficient.  This only starts to matter in the hundreds or thousands of billboards.

Obviously I'd intended to extend ScreenObjects to 2D objects.  But it hasn't happened.  The best way to do the same thing is to make separate ScreenLabels and ScreenMakers with offsets.

**Comment by Sebboulnois** [comment]
Thank you. I tried to add a string to a MaplyScreenObject with the following line:

self.pictogramId.add("KMIA", font: UIFont(name: "Georgia", size: 1), color: UIColor.white)

It seems to work but gives me a gross rendering as follows:

![unadjustednonraw_thumb_1173](https://user-images.githubusercontent.com/15698398/30042174-b2b77fd8-91bd-11e7-9306-b3f936fd98d8.jpg)

I started decreasing the size property in UIFont. However, the text completely disappears when the size is <1 although UIFont allows for size values >0.0. 

Any advice?

Sebastien
